aixplain/factories/metric_factory.py

```diff
@@ -66,62 +73,75 @@ def get(cls, metric_id: Text) -> Metric:
-        #     else:
-        #         headers = {"Authorization": f"Token {cls.api_key}", "Content-Type": "application/json"}
-        #     logging.info(f"Start service for GET Metric  - {url} - {headers}")
-        #     r = _request_with_retry("get", url, headers=headers)
-        #     resp = r.json()
-        #     metric = cls._create_metric_from_response(resp)
-        # except Exception as e:
-        #     if resp is not None and "statusCode" in resp:
-        #         status_code = resp["statusCode"]
-        #         message = resp["message"]
-        #         message = f"Metric Creation: Status {status_code} - {message}"
-        #     else:
-        #         message = "Metric Creation: Unspecified Error"
-        #     logging.error(message)
-        #     raise Exception(f"Status {status_code}: {message}")
-        # return metric
-        raise NotImplementedError("Metric functions are coming soon on the SDK.")
 
-    @classmethod
-    def create_asset_from_id(cls, metric_id: Text) -> Metric:
-        warn(
-            'This method will be deprecated in the next versions of the SDK. Use "get" instead.',
-            DeprecationWarning,
-            stacklevel=2,
-        )
-        return cls.get(metric_id)
+        resp, status_code = None, 200
+        try:
+            url = urljoin(cls.backend_url, f"sdk/metrics/{metric_id}")
+            if cls.aixplain_key != "":
+                headers = {"x-aixplain-key": f"{cls.aixplain_key}", "Content-Type": "application/json"}
+            else:
+                headers = {"Authorization": f"Token {cls.api_key}", "Content-Type": "application/json"}
+            logging.info(f"Start service for GET Metric  - {url} - {headers}")
+            r = _request_with_retry("get", url, headers=headers)
+            resp = r.json()
+            metric = cls._create_metric_from_response(resp)
+        except Exception as e:
+            if resp is not None and "statusCode" in resp:
+                status_code = resp["statusCode"]
+                message = resp["message"]
+                message = f"Metric Creation: Status {status_code} - {message}"
+            else:
+                message = "Metric Creation: Unspecified Error"
+            logging.error(f"Metric Creation Failed: {e}")
+            raise Exception(f"Status {status_code}: {message}")
+        return metric
 
     @classmethod
-    def list_assets(cls, task: Text) -> List[Metric]:
-        """Get list of supported metrics for a given task
+    def list(cls, model_id: Text=None, source_required: bool=None, reference_required: bool=None, page_number: int = 0, page_size: int = 20,) -> List[Metric]:
+        """Get list of supported metrics for the given filters
 
         Args:
-            task (Text): Task to get metric for
+            model_id (Text, optional): ID of model for which metric is to be used. Defaults to None.
+            source_required (bool, optional): Should the metric use source. Defaults to None.
```

Comment:
for boolean variables, I suggest the default values to be `True` or `False`. But if it is None, use the `Optional` typing: `Optional[is_source_required]`